File: emby-lib/src/main/java/us/nineworlds/serenity/emby/model/Stream.java
package us.nineworlds.serenity.emby.model;

import us.nineworlds.serenity.common.media.model.IStream;

public class Stream implements IStream {

  private int id;
  private int streamType;
  private String codec;
  private String index;
  private String channels;
  private String duration;
  private String bitrate;
  private String bitrateMode;
  private String profile;
  private String optimizedForStreaming;
  private String format;
  private String key;
  private String language;
  private String languageCode;

  @Override public String getLanguage() {
    return language;
  }

  @Override public void setLanguage(String language) {
    this.language = language;
  }

  @Override public String getLanguageCode() {
    return languageCode;
  }

  @Override public void setLanguageCode(String languageCode) {
    this.languageCode = languageCode;
  }

  @Override public int getId() {
    return id;
  }

  @Override public void setId(int id) {
    this.id = id;
  }

  @Override public int getStreamType() {
    return streamType;
  }

  @Override public void setStreamType(int streamType) {
    this.streamType = streamType;
  }

  @Override public String getCodec() {
    return codec;
  }

  @Override public void setCodec(String codec) {
    this.codec = codec;
  }

  @Override public String getIndex() {
    return index;
  }

  @Override public void setIndex(String index) {
    this.index = index;
  }

  @Override public String getChannels() {
    return channels;
  }

  @Override public void setChannels(String channels) {
    this.channels = channels;
  }

  @Override public String getDuration() {
    return duration;
  }

  @Override public void setDuration(String duration) {
    this.duration = duration;
  }

  @Override public String getBitrate() {
    return bitrate;
  }

  @Override public void setBitrate(String bitrate) {
    this.bitrate = bitrate;
  }

  @Override public String getBitrateMode() {
    return bitrateMode;
  }

  @Override public void setBitrateMode(String bitrateMode) {
    this.bitrateMode = bitrateMode;
  }

  @Override public String getProfile() {
    return profile;
  }

  @Override public void setProfile(String profile) {
    this.profile = profile;
  }

  @Override public String getOptimizedForStreaming() {
    return optimizedForStreaming;
  }

  @Override public void setOptimizedForStreaming(String optimizedForStreaming) {
    this.optimizedForStreaming = optimizedForStreaming;
  }

  @Override public String getFormat() {
    return format;
  }

  @Override public void setFormat(String format) {
    this.format = format;
  }

  @Override public String getKey() {
    return key;
  }

  @Override public void setKey(String key) {
    this.key = key;
  }
}

File: jellyfin-lib/src/main/kotlin/us/nineworlds/serenity/jellyfin/model/Stream.kt
package us.nineworlds.serenity.jellyfin.model

import us.nineworlds.serenity.common.media.model.IStream

class Stream : IStream {
    private var _id: Int = 0
    private var _streamType: Int = 0
    private var _codec: String? = null
    private var _index: String? = null
    private var _channels: String? = null
    private var _duration: String? = null
    private var _bitrate: String? = null
    private var _bitrateMode: String? = null
    private var _profile: String? = null
    private var _optimizedForStreaming: String? = null
    private var _format: String? = null
    private var _key: String? = null
    private var _language: String? = null
    private var _languageCode: String? = null

    override fun getLanguage(): String? = _language

    override fun setLanguage(language: String?) {
        _language = language
    }

    override fun getLanguageCode(): String? = _languageCode

    override fun setLanguageCode(languageCode: String?) {
        _languageCode = languageCode
    }

    override fun getId(): Int = _id

    override fun setId(id: Int) {
        _id = id
    }

    override fun getStreamType(): Int = _streamType

    override fun setStreamType(streamType: Int) {
        _streamType = streamType
    }

    override fun getCodec(): String? = _codec

    override fun setCodec(codec: String?) {
        _codec = codec
    }

    override fun getIndex(): String? = _index

    override fun setIndex(index: String?) {
        _index = index
    }

    override fun getChannels(): String? = _channels

    override fun setChannels(channels: String?) {
        _channels = channels
    }

    override fun getDuration(): String? = _duration

    override fun setDuration(duration: String?) {
        _duration = duration
    }

    override fun getBitrate(): String? = _bitrate

    override fun setBitrate(bitrate: String?) {
        _bitrate = bitrate
    }

    override fun getBitrateMode(): String? = _bitrateMode

    override fun setBitrateMode(bitrateMode: String?) {
        _bitrateMode = bitrateMode
    }

    override fun getProfile(): String? = _profile

    override fun setProfile(profile: String?) {
        _profile = profile
    }

    override fun getOptimizedForStreaming(): String? = _optimizedForStreaming

    override fun setOptimizedForStreaming(optimizedForStreaming: String?) {
        _optimizedForStreaming = optimizedForStreaming
    }

    override fun getFormat(): String? = _format

    override fun setFormat(format: String?) {
        _format = format
    }

    override fun getKey(): String? = _key

    override fun setKey(key: String?) {
        _key = key
    }
}
